core/src/com/mygdx/game/steerers/FollowPathSteerer.java

```diff
@@ -136,21 +206,61 @@ public void stopSteering () {
 	public void finishSteering () {
 		pathToRender.clear();
 	}
-	
+
+	boolean deadlockDetection;
+	float deadlockDetectionStartTime;
+	float collisionDuration;
+	static final float deadlockTime = .5f;
+	static final float maxNoCollisionTime = deadlockTime + .5f;
+
 	@Override
 	public void onSteering () {
+		
+		if (prioritySteering.getSelectedBehaviorIndex() == 0) {
+				float pr = proximity.getRadius() * 1.5f;
+				if (linePath.getEndPoint().dst2(steerableBody.getPosition()) <= pr * pr) {
+					// Disable collision avoidance near the end of the path since the obstacle 
+					// will likely prevent the entity from reaching the target.
+					collisionAvoidanceSB.setEnabled(false);
+					deadlockDetectionStartTime = Float.POSITIVE_INFINITY;
+				}
+				else if (deadlockDetection) {
+					// Accumulate collision time during deadlock detection
+					collisionDuration += GdxAI.getTimepiece().getDeltaTime();
+
+					if (GdxAI.getTimepiece().getTime() - deadlockDetectionStartTime > deadlockTime && collisionDuration > deadlockTime *.6f) {
+						// Disable collision avoidance since most of the deadlock detection period has been spent on collision avoidance
+						collisionAvoidanceSB.setEnabled(false);
+					}
+				}
+				else {
+					// Start deadlock detection
+					deadlockDetectionStartTime = GdxAI.getTimepiece().getTime();
+					collisionDuration = 0;
+					deadlockDetection = true;
+				}
+		}
+		else {
+			if (deadlockDetection && !collisionAvoidanceSB.isEnabled() && GdxAI.getTimepiece().getTime() - deadlockDetectionStartTime > maxNoCollisionTime) {
+				collisionAvoidanceSB.setEnabled(true);
+				deadlockDetection = false;
+			}
+		}
+
 		// Check if steering target path segment changed.
 		int traversedSegment = followPathSB.getPathParam().getSegmentIndex();
 		if (traversedSegment > currentSegmentIndex) {
 			// Update model target orientation. Current orientation wi
 			currentSegmentIndex = traversedSegment;
-			Segment<Vector3> segment = linePath.getSegments().get(currentSegmentIndex);
-			steerableBody.setModelTargetOrientation(segment.getEnd().x - segment.getBegin().x, segment.getEnd().z - segment.getBegin().z);
+/*
+//			Segment<Vector3> segment = linePath.getSegments().get(currentSegmentIndex);
+//			steerableBody.setModelTargetOrientation(segment.getEnd().x - segment.getBegin().x, segment.getEnd().z - segment.getBegin().z);
 			// Update current navmesh triangle
 			steerableBody.currentTriangle = navMeshPointPath.getToTriangle(currentSegmentIndex);
 			// Set model to be visible on the same layer as mesh part index of current triangle
 			steerableBody.visibleOnLayers.clear();
 			steerableBody.visibleOnLayers.set(steerableBody.currentTriangle.meshPartIndex);
+ */
```

Comment:
It makes sense to move those things up as far as I can tell. This method of getting current triangle only works for path following anyway, so it makes sense for the calling class to do it.

Using the direction of motion seems fine also, though I have noticed a problem with it. When the steerable is in motion, following a path, and the player clicks the navmesh to set a new path, the steerable will almost always overshoot the new goal point. It then has to backtrack a small bit, which creates a motion in the reverse direction. This would have to be fixed to prevent turning the model around at the end. 
Otherwise, some threshold for speed should be required for model turning to be allowed.